### PR TITLE
Improve the handbook with feedback from the 1.4 Release Retro

### PR DESCRIPTION
**Pull request by kimwnasptd**
I'm augmenting the release handbook to include discussion points from the 1.4 release retro we had  https://github.com/kubeflow/community/blob/master/releases/retrospectives/release-1.4.md 

Points addressed:
1. Release team selection process
2. Encouragement to have docs PRs alongside feature PRs
3. Add more details in the Timeline, regarding RCs
4. Explicitly mention that we should also have a tracking issue for the release

I also think that we should look into breaking the handbook into some smaller files, like `roles`, `timeline` etc. It's currently a huge 300+ lines and will keep getting bigger.

/cc @annajung 
/cc @shannonbradshaw 
/cc @jbottum

### CODE REVIEW

File: releases/handbook.md [hunk 1]
```diff
@@ -155,26 +155,25 @@ The manifests repo will be following the release process below:
...
-| 17 | Release |
+| **Week** | **Who** | **What** |
+| -------- | ------- | -------- |
+| Week 0 | Release Manager | [Preparation](#preparation) |
+| Week 1 | Release Manager | Start of Release Cycle |
+| Week 1 | Community | [Development](#development-10-weeks) |
+| Week 10 | Release Team | [Feature Freeze](#feature-freeze-2-weeks) |
+| Week 10 | Manifest WG | [rc.0 Released](#feature-freeze-2-weeks) |
+| Week 10 | Docs Lead | [Documentation](#documentation) |
+| Week 10 | Manifest WG | [rc.1 Released](#feature-freeze-2-weeks) |
```
Comment (annajung) [inline releases/handbook.md:166]: According to the [handbook](https://github.com/kubeflow/community/blob/master/releases/handbook.md#feature-freeze-2-weeks), I believe rc.1 is released 2 weeks after the feature freeze? 

I think rc.1 should happen in week 12 since there is a schedule for rc.0 during week 10 already.

File: releases/handbook.md [hunk 2]
```diff
@@ -194,15 +193,38 @@ Criteria for timeline that the team needs to consider
 - Kubecon dates - let’s not hard block on events, but keep them in mind since we know community members might get doublebooked.
 - Associated events (aka. AI Day at Kubecon, Tensorflow events) - we want to keep them in mind.
 
-**Success Criteria:** Release team selected, schedule sent to kubeflow-discuss, all release team members have the proper permissions and are meeting regularly.
+**Success Criteria:**
+* Release team selected
+* Schedule sent to kubeflow-discuss
```
Comment (annajung) [inline releases/handbook.md:198]: Can we rephrase this to "send out release timeline proposal PR to kubeflow-discuss to receive feedback from the community"? and maybe during the week 1 phase, we have success criteria of "Send out the finalized release schedule to kubeflow-discuss"?

File: releases/handbook.md [hunk 3]
```diff
@@ -194,15 +193,38 @@ Criteria for timeline that the team needs to consider
...
+* All release team members have the proper permissions and are meeting regularly
```
Comment (annajung) [inline releases/handbook.md:199]: I think it would be helpful to list where the new release team members need to be added for permission. For example, the release GitHub team needs to be updated https://github.com/kubeflow/internal-acls/pull/511 and you mentioned a new member need to join kubeflow-discuss mailing group. Can we add those here and is there anything else that we can specify here?

File: releases/handbook.md (hunk 1 above)
Comment (kimwnasptd) [reply releases/handbook.md:166]: Nice catch! Yes you are right, it should happen on week 12 which is before the Manifests Testing starts.

I think I'll need to rephrase this sentence in the [Feature Freeze (handbook)](https://github.com/kubeflow/community/blob/master/releases/handbook.md#feature-freeze-2-weeks)
> On the last day of feature freeze, cut a new vX.Y-rc.1 RC tag on the vX.Y-branch release branch.

To make it clear that:
1. `Feature Freeze` does not stop at that time, it continues through the subsequent phases
2. We cut the RC1 because we are preparing for `Manifests Testing`, and not because `Feature Freeze` stops [which doesn't happen]. We want the Manifests leads to use an RC before they start their testing

Comment (kimwnasptd) [reply releases/handbook.md:166]: Oh and also I think we'll need to fix the week 10 -> week 12 in this one as well
https://github.com/kubeflow/community/tree/master/releases/release-1.5#timeline

I'll push a commit with this as well

File: releases/handbook.md (hunk 3 above)
Comment (kimwnasptd) [reply releases/handbook.md:199]: I've mentioned that everyone needs to be part of the `internal-acl` repo, although it's a hyperlink, in the list of checkboxes for the [preparation phase](https://github.com/kubeflow/community/blob/master/releases/handbook.md#preparation).

I'll make it more explicit by adding sub-checks in that step to note that release team mambers need to be:
1. In the internal-acl repo and github team
2. Part of the `kubeflow-discuss` mailing list, in order to have access to the assets (drive, notes, recordings) of the release team

WDYT?

File: releases/handbook.md (hunk 2 above)
Comment (kimwnasptd) [reply releases/handbook.md:198]: Since there are the following checkboxes in the [preparation phase](https://github.com/kubeflow/community/blob/master/releases/handbook.md#preparation) above the success criteria:
> [ ] Publish draft schedule to kubeflow-discuss, with actual dates
> [ ] Get lazy consensus on the release schedule from the WG leads

WDYT about just replacing `Schedule sent to kubeflow-discuss` with your proposed `Send out the finalized release schedule to kubeflow-discuss` sentence there?

Currently the Preparation phase does not have a deadline, but from your comment about week 1 I understand you'd like to start defining this phase more concretely?

File: releases/handbook.md (hunk 1 above)
Comment (annajung) [reply releases/handbook.md:166]: That sounds good!

File: releases/handbook.md (hunk 2 above)
Comment (annajung) [reply releases/handbook.md:198]: Ah okay, if it's already in the preparation phase I think it's okay to leave it as it is. No need to rephrase. I wanted to make sure that the release timeline was proposed to the community for feedback before finalizing. :)

File: releases/handbook.md (hunk 3 above)
Comment (annajung) [reply releases/handbook.md:199]: again, totally missed the preparation phase. 

I think the only item missing is being part of the `kubeflow-discuss`. I think just adding that should be sufficient!

File: releases/handbook.md (hunk 1 above)
Comment (annajung) [reply releases/handbook.md:166]: I believe https://github.com/kubeflow/community/tree/master/releases/release-1.5#timeline will be fixed with https://github.com/kubeflow/community/pull/538.
